rfcs/text/0000-dss-deletion-daemon.md

```diff
@@ -0,0 +1,268 @@
+   the data is Permanently deleted.
+
+5) Data that has been physical deleted remain in the bucket for 7 days before being permanently deleted from the bucket.
+   This grace period allows the deletion to be reversed within that time period. After the grace period the data will 
+   be permanently deleted.
+   
+### Primary indexes
+
+Bundle tombstones are indexed verbatim by the Data Store. The JSON schema
+for deletion markers is defined such that there are no conflicts with the
+top-level keys in the index documents for regular bundles (`files`, `manifest`,
+`uuid`, `version` etc). Bundle tombstones are therefore compatible with
+regular bundle documents with respect to in the same index even if
+Elasticsearch's dynamic mapping is enabled.
+
+### Replication
+
+When a bundle is deleted it is removed from all replicas.
+
+### Consistency
+
+It is the administrator's responsibility to enumerate all bundles referring to
+a file that is to be deleted and issue a delete requests for each of
+them. If the administrator misses such a bundle (a *dangling bundle*), users
+requesting `GET /bundle` for a dangling bundle will be able to retrieve its
+manifest. Likewise, `POST /search` requests will return the metadata for
+dangling bundles. Secondary indexes like the Data Browser (Orange Box) will
+continue to hand out (meta)data for dangling bundles. If users request `GET
+/files` for a file contained in a dangling bundle and the requested file was
+referenced by another already logically (physically) deleted bundle, the Data
+Store will respond with a 404 (500) status code.
+
+The administrator can initiate a consistency check against a storage replica to
+list dangling bundles. The administrator can then issue more `DELETE /bundles`
+requests for those dangling bundles.
+
+### Secondary indexes
+
+The Data Store can optionally notify subscribers about the creation of a bundle
+tombstone. Secondary index services need to register a subscription using
+a query that matches deletion markers. When they are notified about a deletion
+marker, they need to take the necessary actions to remove the data from their
+database or index.
+
+### Derived data
+
+The metadata in bundles containing derived data must reference the input bundle
+versions. Administrators are responsible for tracking down data bundles with
+data derived from data in deleted bundles and to issue deletion requests for
+them.
+
+### Tiered storage
+
+Physical deletion of blobs is effective across all tiered storage classes in
+the underlying blob store. Additional cost may be incurred for deleting data recently moved to a
+slower access storage tier. Deletion occur There is no additional delays cause when deleting from a 
+tiered storage system such as AWS glacier.
+
+### Undoing logical deletions
+
+A bundle can be restored using `restore_bundle.py <uuid.<version>` if has not been physically deleted. This process
```

Comment:
This is a DCP service RFC, and as such, I think it should specify the semantics of HTTP REST/RPC APIs, not details of Python scripts. The deletions semantics described in this RFC should be available in an API, not through a DSS service admin "backdoor" defined by a Python script.